### PR TITLE
simple sd_save flag

### PR DESCRIPTION
**Pull request by Kronos2308**
something that allows you to decide which title to redirect games saves to sd and which does not

### CODE REVIEW

**Comment by SciresM** [comment]
Please fix whitespace, and rename flag to "redirect_save".

**Comment by Kronos2308** [comment]
sure why not

**Comment by Kronos2308** [comment]
I think it is ready

**Comment by hippydave** [comment]
Looks like you need to un-indent the "if".

File: stratosphere/ams_mitm/source/fs_mitm/fsmitm_service.cpp [hunk 1]
```diff
@@ -162,10 +162,15 @@ Result FsMitmService::OpenFileSystemWithId(Out<std::shared_ptr<IFileSystemInterf
 }
 
 Result FsMitmService::OpenSaveDataFileSystem(Out<std::shared_ptr<IFileSystemInterface>> out_fs, u8 space_id, FsSave save_struct) {
+    const bool has_redirect_save_flags = Utils::HasFlag(this->title_id, "redirect_save");
     bool should_redirect_saves = false;
     if (R_FAILED(Utils::GetSettingsItemBooleanValue("atmosphere", "fsmitm_redirect_saves_to_sd", &should_redirect_saves))) {
         return ResultAtmosphereMitmShouldForwardToSession;
     }
+	//check if flag exist
+	if (!has_redirect_save_flags) {
```
Comment (mailwl) [inline stratosphere/ams_mitm/source/fs_mitm/fsmitm_service.cpp]: `if` is shifted right compared to closed curly brace

**Comment by Kronos2308** [comment]
yep i see, ignore the system settings.ini

**Comment by Kronos2308** [comment]
now check if the save_redirection is enabled or disabled in system_settings.ini, only if is enable watch for the redirect_save.flag file

**Comment by Kronos2308** [comment]
I do not see another error, it should be good now, correct me if I'm wrong

File: stratosphere/ams_mitm/source/fs_mitm/fsmitm_service.cpp [hunk 2]
```diff
@@ -163,12 +163,13 @@ Result FsMitmService::OpenFileSystemWithId(Out<std::shared_ptr<IFileSystemInterf
...
 Result FsMitmService::OpenSaveDataFileSystem(Out<std::shared_ptr<IFileSystemInterface>> out_fs, u8 space_id, FsSave save_struct) {
     bool should_redirect_saves = false;
+    const bool has_redirect_save_flags = Utils::HasFlag(this->title_id, "redirect_save");
     if (R_FAILED(Utils::GetSettingsItemBooleanValue("atmosphere", "fsmitm_redirect_saves_to_sd", &should_redirect_saves))) {
         return ResultAtmosphereMitmShouldForwardToSession;
     }
 
-    /* For now, until we're sure this is robust, only intercept normal savedata. */
-    if (!should_redirect_saves || save_struct.SaveDataType != FsSaveDataType_SaveData) {
+    /* For now, until we're sure this is robust, only intercept normal savedata , checkk if flag exist*/
```
Comment (mailwl) [inline stratosphere/ams_mitm/source/fs_mitm/fsmitm_service.cpp]: 'checkk'

**Comment by Kronos2308** [comment]
jajajaj

File: stratosphere/ams_mitm/source/fs_mitm/fsmitm_service.cpp (hunk 2 above)
Comment (Kronos2308) [reply stratosphere/ams_mitm/source/fs_mitm/fsmitm_service.cpp]: done

**Pull request merged**